doc/source/whatsnew/v1.0.0.rst

```diff
@@ -165,6 +165,7 @@ Plotting
 
 - Bug in :meth:`Series.plot` not able to plot boolean values (:issue:`23719`)
 -
+- Bug in :meth:`DataFrame.plot` when ``kind='box'`` and data contains datatime column (:issue:`22799`)
```

Comment:
```suggestion
- Bug in :meth:`DataFrame.plot` when ``kind='box'`` and data contains datetime or timedelta data. These types are now automatically dropped (:issue:`22799`)
```

Reply:
Thanks for taking a look! Updated! @TomAugspurger